### ISSUE TITLE
[Question] Start DSCM when starting Dark souls

### ISSUE THREAD

**Issue by gpotter2** [Wulf2k/DaS-PC-MPChan#95]
I might have found a way to auto start DSCM when starting dark souls from steam...

Is it a good idea ?

**Comment by Chronial** [comment]
That depends how this is done. If this needs modifications to the system, then it's probably not a good idea.

**Comment by gpotter2** [comment]
What i do is just add a folder in the "Dark Souls Prepare to Die Edition\REDIST" folder called "GWFL", then inside it a "Gfwlivesetup.exe" file which is a fake one and only start a DSCM.exe file in the same folder with an "--steam-autostarted" arg.

Then all you need to do is set the "gfwlivesetup" var of the registery to 0 and it will start the game next time dark souls starts

That's not that messy :)

**Comment by Chronial** [comment]
That is indeed a quite nice idea, but I'm not quite sure that the benefits justify the hackishnes.

**Comment by gpotter2** [comment]
I'm gonna make a working code and publish it as pull request so we can discuss of it...

**Comment by Chronial** [comment]
Note that any such change is blocked by #94, which in turn is probably lower priority than #92

**Comment by gpotter2** [comment]
Ok then :) I can wait

**Comment by gpotter2** [comment]
Ok i finally got a working code.

It will actually ask once when you first want to enable the feature to go to admin mode in order to change some permissions in the registry (they are not dangerous or anything, and that's the only way you can get through steam installation for the first time... without patching the dark souls files like DsFix). After that it won't ask for it anymore...

**Comment by Chronial** [comment]
I thought about this again, and changed my mind – I really don't think this is a good idea.

1. Changing the registry value back is horrible – that means I will get those "steam wants admin access" popups again every time I start Dark Souls. That is not an improvement over starting DSCM first – it's actually worse.
2. Even if the user didn't have the correct registry setting anyway, they will now associate that annoying popup with DSCM, which will cause a lot of annoying feedback.
3. Won't this start DSCM with administrator access? I am not a fan of that.

**Comment by gpotter2** [comment]
Not at all !

Having the popup each time would be a way too enoying ! 

It only occures once in DSCM life (not each time you restart it...): First time you want to enable the feature, it prompt the popup to ask to get admin. 

Then what it does is changing the permissions of the registery key i want to access (which is only for darksouls) so you are able to use this key without being admin.

But in order to change the permissions the first time, you need to be admin, that's why it's asking for it once... 

After that, you may disable the function, restart dark souls / dscm or whatever it won't ask for admin...

> that means I will get those "steam wants admin access"
What do you mean by that ?

**Comment by Chronial** [comment]
>> that means I will get those "steam wants admin access"

> What do you mean by that ?

The popup that you get everytime you start dark souls as long as `gfwlivesetup` is not `1`:

![UAC](http://i.stack.imgur.com/9bZNB.jpg)

**Comment by gpotter2** [comment]
Huh... I didn't though about that ... I'm gonna look for a way to disable it...

**Comment by gpotter2** [comment]
Well i've found a completly different way of doing it :P

Can we just replace DARKSOULS.exe by a fake one which start the real darksouls.exe + DSCM ?

It's working much easier this way :smile:

**Comment by Wulf2k** [comment]
No.  It would be overwritten every time Steam verified files.

Also, again, needs administrator

**Comment by gpotter2** [comment]
Well, i have no more ideas...
